### PR TITLE
set-certificate: add `sync' prop

### PR DESCRIPTION
**Pull request by Amygos**
- set-certificate: show certificate state as output
- tests: check set-certificate output
- set-certificate: add sync flag
- set-certificate: add `sync_timeout` option
- set-certificate: show operation progress

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,14 +178,21 @@ It can be used when there is no hostname (or hostname + path) route
 configured on traefik module or if the service is not make accessible via
 traefik.
 
-The action takes 1 parameter:
+The action takes 3 parameter:
```
Comment (andre8244) [inline README.md:181]:
```suggestion
The action takes 3 parameters:
```

File: README.md [hunk 2]
```diff
@@ -178,14 +178,21 @@ It can be used when there is no hostname (or hostname + path) route
...
 - `fqdn`: the fqdn of the requested certificate
+- `sync`: wait until the certificate is obtained before return, default `false`.
+- `sync_timeout`: Max number of seconds to wait for the certificate to be obtined, default `120`.
```
Comment (andre8244) [inline README.md:184]:
```suggestion
- `sync_timeout`: Max number of seconds to wait for the certificate to be obtained, default `120`.
```

File: imageroot/actions/set-certificate/validate-input.json [hunk 3]
```diff
@@ -17,6 +17,17 @@
             "type":"string",
             "format": "hostname",
             "title": "A fully qualified domain name"
-        }
+        },
+	"sync": {
+	    "type":"boolean",
+	    "description": "Wait for the certificate to be obtined before return"
```
Comment (andre8244) [inline imageroot/actions/set-certificate/validate-input.json:23]:
```suggestion
	    "description": "Wait for the certificate to be obtained before return"
```

File: imageroot/actions/set-certificate/validate-input.json [hunk 4]
```diff
@@ -17,6 +17,17 @@
...
+	},
+	"sync_timeout": {
+	    "type":"integer",
+	    "minimum": 1,
+	    "description": "Max number of seconds to wait for the certificate to be obtined",
```
Comment (andre8244) [inline imageroot/actions/set-certificate/validate-input.json:28]:
```suggestion
	    "description": "Max number of seconds to wait for the certificate to be obtained",
```

**Review by andre8244 — COMMENTED** [review]
Just a few typos

File: imageroot/actions/set-certificate/validate-input.json [hunk 5]
```diff
@@ -17,6 +17,17 @@
             "type":"string",
             "format": "hostname",
             "title": "A fully qualified domain name"
-        }
+        },
+	"sync": {
```
[the remark this answers is not in the document]
Comment (Amygos) [reply imageroot/actions/set-certificate/validate-input.json:21]: :+1:

**Pull request merged**